node/node.go

```diff
@@ -615,86 +615,85 @@ func (n *Node) createConsensusTopic(messageProcessor p2p.MessageProcessor) error
 	return n.messenger.RegisterMessageProcessor(n.consensusTopic, messageProcessor)
 }
 
-// SendTransaction will send a new transaction on the topic channel
-func (n *Node) SendTransaction(
-	nonce uint64,
-	senderHex string,
-	receiverHex string,
-	value string,
-	gasPrice uint64,
-	gasLimit uint64,
-	transactionData []byte,
-	signature []byte) (string, error) {
-
-	if n.shardCoordinator == nil || n.shardCoordinator.IsInterfaceNil() {
-		return "", ErrNilShardCoordinator
+// SendBulkTransactions sends the provided transactions as a bulk, optimizing transfer between nodes
+func (n *Node) SendBulkTransactions(txs []*transaction.Transaction) (uint64, error) {
+	if len(txs) == 0 {
+		return 0, ErrNoTxToProcess
 	}
 
-	sender, err := n.addrConverter.CreateAddressFromHex(senderHex)
-	if err != nil {
-		return "", err
+	filteredTransactions := n.filterTransactions(txs)
+	if len(filteredTransactions) == 0 {
+		return 0, fmt.Errorf("%w after validating", ErrNoTxToProcess)
 	}
 
-	receiver, err := n.addrConverter.CreateAddressFromHex(receiverHex)
-	if err != nil {
-		return "", err
-	}
+	n.addTransactionsToSendPipe(filteredTransactions)
 
-	senderShardId := n.shardCoordinator.ComputeId(sender)
+	return uint64(len(filteredTransactions)), nil
+}
 
-	valAsBigInt, ok := big.NewInt(0).SetString(value, 10)
-	if !ok {
-		return "", ErrInvalidValue
+func (n *Node) filterTransactions(txs []*transaction.Transaction) []*transaction.Transaction {
+	filteredTransactions := make([]*transaction.Transaction, 0, len(txs))
+	var err error
+	for _, tx := range txs {
+		err = n.validateTx(tx)
+		if err != nil {
+			continue
+		}
+
+		filteredTransactions = append(filteredTransactions, tx)
 	}
 
-	tx := transaction.Transaction{
-		Nonce:     nonce,
-		Value:     valAsBigInt,
-		RcvAddr:   receiver.Bytes(),
-		SndAddr:   sender.Bytes(),
-		GasPrice:  gasPrice,
-		GasLimit:  gasLimit,
-		Data:      transactionData,
-		Signature: signature,
+	return filteredTransactions
+}
+
+func (n *Node) addTransactionsToSendPipe(txs []*transaction.Transaction) {
+	if check.IfNil(n.txAcumulator) {
+		log.Error("node has a nil tx accumulator instance")
+		return
 	}
 
-	err = n.validateTx(&tx)
-	if err != nil {
-		return "", err
+	for _, tx := range txs {
+		n.txAcumulator.AddData(tx)
 	}
+}
 
-	txBuff, err := n.marshalizer.Marshal(&tx)
-	if err != nil {
-		return "", err
+func (n *Node) setTxAccumulator(txAccumulator Accumulator) error {
+	if check.IfNil(txAccumulator) {
+		return ErrNilTxAccumulator
 	}
+	n.txAcumulator = txAccumulator
 
-	txHexHash := hex.EncodeToString(n.hasher.Compute(string(txBuff)))
+	go func() {
```

Comment:
make a separate method for all under go func

Reply:
done